databricks/sdk/errors/deserializer.py

```diff
@@ -0,0 +1,100 @@
+import abc
+import json
+import logging
+import re
+from typing import Optional
+
+import requests
+
+
+class _ErrorDeserializer(abc.ABC):
+    """A parser for errors from the Databricks REST API."""
+
+    @abc.abstractmethod
+    def parse_error(self, response: requests.Response, response_body: bytes) -> Optional[dict]:
+        """Parses an error from the Databricks REST API. If the error cannot be parsed, returns None."""
+
+
+class _EmptyDeserializer(_ErrorDeserializer):
+    """A parser that handles empty responses."""
+
+    def parse_error(self, response: requests.Response, response_body: bytes) -> Optional[dict]:
+        if len(response_body) == 0:
+            return {'message': response.reason}
+        return None
+
+
+class _StandardErrorDeserializer(_ErrorDeserializer):
+    """
+    Parses errors from the Databricks REST API using the standard error format.
+    """
+
+    def parse_error(self, response: requests.Response, response_body: bytes) -> Optional[dict]:
+        try:
+            payload_str = response_body.decode('utf-8')
+            resp: dict = json.loads(payload_str)
+        except json.JSONDecodeError as e:
+            logging.debug('_StandardErrorParser: unable to deserialize response as json', exc_info=e)
+            return None
+
```

Comment:
My understanding is that the type checking is not enforced and would thus result in an error at line 41 if `resp` is not a dictionary. Should we explicitly check that it is a `dict`?

```suggestion
        try:
            payload_str = response_body.decode('utf-8')
        except json.JSONDecodeError as e:
            logging.debug('_StandardErrorParser: unable to deserialize response as json', exc_info=e)
            return None
            
        resp = json.loads(payload_str)
        if not isinstance(resp, dict):
            logging.debug('_StandardErrorParser: json response should be a json message', exc_info=e)
            return None

```